ivy/data_classes/array/experimental/manipulation.py

```diff
@@ -1134,6 +1134,62 @@ def take(
             self, indices, axis=axis, mode=mode, fill_value=fill_value, out=out
         )
 
+    def unflatten(
+        self: Union[int, ivy.Array, ivy.NativeArray],
```

Comment:
In this case, the array object is calling this function therefore, "self" can only be of type "Ivy.Array" to fit accordingly to the logic of the Array class. Please refer to the previous code you can find in this file and the documentation.

Reply:
Corrected 👍